File: models/Gpt-3.5-turbo/py/code/advanced_techniques/One_Bit_Positions.py

def count_pairs(binary_string):
    n = len(binary_string)
    one_bits = [0] * n
    count = 0
    
    # Calculate the number of one-bits at each position
    for i in range(n):
        if binary_string[i] == '1':
            count += 1
        one_bits[i] = count
    
    # Count the number of pairs for each distance k
    pairs = [0] * (n-1)
    for k in range(1, n):
        for i in range(n-k):
            if binary_string[i] == '1' and binary_string[i+k] == '1':
                pairs[k-1] += 1
    
    return pairs

# Read the binary string from input
binary_string = input().strip()

# Call the function and print the result
result = count_pairs(binary_string)
print(*result)

File: models/Gpt-3.5-turbo/cpp/code/problems/advanced_techniques/One_Bit_Positions.cpp

#include <iostream>
#include <vector>
using namespace std;

int main() {
    string s;
    cin >> s;
    int n = s.length();

    vector<int> count(n, 0);
    int oneCount = 0;

    for (int i = 0; i < n; i++) {
        if (s[i] == '1') {
            oneCount++;
        }
        count[i] = oneCount;
    }

    vector<int> result(n-1, 0);

    for (int k = 1; k < n; k++) {
        for (int i = 0; i < n-k; i++) {
            if (s[i] == '1' && s[i+k] == '1') {
                result[k-1]++;
            }
        }
    }

    for (int i = 0; i < n-1; i++) {
        cout << result[i] << " ";
    }

    return 0;
}
